File: vbox-app/src/main/java/com/kedzie/vbox/api/IDisplay.java
package com.kedzie.vbox.api;

import java.util.Map;

import com.kedzie.vbox.soap.KSOAP;

/**
 * The {@link IDisplay} interface represents the virtual machine's display. 
 * <p>The object implementing this interface is contained in each {@link IConsole#getDisplay}<b></b> attribute and represents the visual output of the virtual machine.</p>
 * <p>The virtual display supports pluggable output targets represented by the {@link IFramebuffer} interface. Examples of the output target are a window on the host computer or an RDP session's display on a remote computer.</p>
 */
@KSOAP
public interface IDisplay extends IManagedObjectRef {

    @KSOAP public Map<String, String> getScreenResolution(@KSOAP(type="unsignedInt", value="screenId") int screenId);

    @KSOAP public byte[] takeScreenShotToArray(@KSOAP(type="unsignedInt", value="screenId") int screenId, @KSOAP(type="unsignedInt", value="width") int width, @KSOAP(type="unsignedInt", value="height") int height);

    @KSOAP public byte[] takeScreenShotPNGToArray(@KSOAP(type="unsignedInt", value="screenId") int screenId, @KSOAP(type="unsignedInt", value="width") int width, @KSOAP(type="unsignedInt", value="height") int height);
	
}

File: vbox-app/src/main/java/com/kedzie/vbox/api/IDisplay.kt
package com.kedzie.vbox.api

import com.kedzie.vbox.api.jaxb.BitmapFormat
import com.kedzie.vbox.soap.Ksoap
import com.kedzie.vbox.soap.KsoapProxy

/**
 * The [IDisplay] interface represents the virtual machine's display.
 *
 * The object implementing this interface is contained in each [IConsole.getDisplay]**** attribute and represents the visual output of the virtual machine.
 *
 * The virtual display supports pluggable output targets represented by the [IFramebuffer] interface. Examples of the output target are a window on the host computer or an RDP session's display on a remote computer.
 */
@KsoapProxy
@Ksoap
interface IDisplay : IManagedObjectRef {

    suspend fun getScreenResolution(@Ksoap(type = "unsignedInt", value = "screenId") screenId: Int):
            Map<String, String>

    suspend fun takeScreenShotToArray(@Ksoap(type = "unsignedInt", value = "screenId") screenId: Int,
                              @Ksoap(type = "unsignedInt", value = "width") width: Int,
                              @Ksoap(type = "unsignedInt", value = "height") height: Int,
                              bitmapFormat: BitmapFormat): ByteArray
}
